lib/langchain/loader.rb

```diff
@@ -91,7 +91,11 @@ def load(&block)
 
     def load_from_url
       unescaped_url = URI.decode_www_form_component(@path)
-      escaped_url = URI::DEFAULT_PARSER.escape(unescaped_url)
+
+      # Ensure compatibility with `URI::RFC2396_PARSER` (Ruby 3.4+) and `URI::DEFAULT_PARSER` (<= 3.3).
+      uri_parser = defined?(URI::RFC2396_PARSER) ? URI::RFC2396_PARSER : URI::DEFAULT_PARSER
```

Comment:
Instead of doing this in a few different places -- it would probably make sense to define it as a Util and obfuscate the actual class used: `URI::RFC2396_PARSER` or `URI::DEFAULT_PARSER`?

Reply:
That makes sense! I opened PR #994.